pkg/k8s/labels.go

```diff
@@ -24,6 +24,26 @@ const (
 	// namespace of the Conduit control plane.
 	ControllerNSLabel = "conduit.io/control-plane-ns"
 
+	// ProxyDeploymentLabel is injected into mesh-enabled apps, identifying the
+	// deployment that this proxy belongs to.
+	ProxyDeploymentLabel = "conduit.io/proxy-deployment"
```

Comment:
these are a bit more verbose than the `deployment` label we had before.  @olix0r has also suggested we use `k8s_deployment` which is a middle ground between `deployment` and `conduit.io/proxy-deployment`.

Reply:
👍 in favour of `k8s_deployment`.

Reply:
Agree these are a bit verbose. Our current implementation prefixes all k8s labels and annotations with `conduit.io/`.

A k8s label key `conduit.io/proxy-deployment` translates to `conduit_io_proxy_deployment` at Prometheus scrape time. We could modify our Prometheus scrape config to instead translate it to `k8s_deployment`, so at least in metrics it shows up as `k8s_deployment`. What do you think?

For reference, our current annotations and labels:
```yaml
template:
  metadata:
    annotations:
      conduit.io/created-by: conduit/cli git-fb1d6a5c
      conduit.io/proxy-version: git-fb1d6a5c
    creationTimestamp: null
    labels:
      conduit.io/control-plane-component: prometheus
      conduit.io/control-plane-ns: conduit
      conduit.io/proxy-deployment: prometheus
```

...and some other example annotations:
```bash
$ kubectl -n kube-system get po -o jsonpath='{.items[*].metadata.annotations}'
map[kubernetes.io/config.source:file scheduler.alpha.kubernetes.io/critical-pod: kubernetes.io/config.hash:1b6fafb5dc39ea18814d9bc27da851eb kubernetes.io/config.mirror:1b6fafb5dc39ea18814d9bc27da851eb kubernetes.io/config.seen:2018-02-09T22:52:27.314281134Z] map[kubernetes.io/config.mirror:7b1d993d9d79998912ae7bae876df519 kubernetes.io/config.seen:2018-02-09T22:52:27.314286935Z kubernetes.io/config.source:file scheduler.alpha.kubernetes.io/critical-pod: kubernetes.io/config.hash:7b1d993d9d79998912ae7bae876df519] map[kubernetes.io/config.seen:2018-02-09T22:52:27.314288483Z kubernetes.io/config.source:file scheduler.alpha.kubernetes.io/critical-pod: kubernetes.io/config.hash:02bf6a87531eb2d995180ff52c8f9419 kubernetes.io/config.mirror:02bf6a87531eb2d995180ff52c8f9419] map[kubernetes.io/config.source:file scheduler.alpha.kubernetes.io/critical-pod: kubernetes.io/config.hash:69c12074e336b0dbbd0a1666ce05226a kubernetes.io/config.mirror:69c12074e336b0dbbd0a1666ce05226a kubernetes.io/config.seen:2018-02-09T22:52:27.314289725Z]
```

Reply:
I agree we should keep the conduit.io/ prefix on things (labels, etc.) we add to users' objects.